forc-plugins/forc-publish/src/error.rs

```diff
@@ -0,0 +1,123 @@
+use reqwest::StatusCode;
+use serde::Deserialize;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("I/O error")]
+    IoError(#[from] std::io::Error),
+
+    #[error("Json error")]
+    JsonError(#[from] serde_json::Error),
+
+    #[error("HTTP error")]
+    HttpError(#[from] reqwest::Error),
+
+    #[error("TOML error")]
+    TomlError(#[from] toml::ser::Error),
+
+    #[error("Failed to get relative path")]
+    RelativePathError(#[from] std::path::StripPrefixError),
+
+    #[error("{error}")]
+    ApiResponseError { status: StatusCode, error: String },
+
+    #[error("Forc.toml not found in the current directory")]
+    ForcTomlNotFound,
+}
+
+#[derive(Deserialize)]
+pub struct ApiErrorResponse {
+    error: String,
+}
+
+impl Error {
+    /// Converts a `reqwest::Response` into an `ApiError`
+    pub async fn from_response(response: reqwest::Response) -> Self {
```

Comment:
Could implement the idiomatic `From<reqwest::Response>` trait; so we get automatic `.into()` for the `response` type.

Reply:
It would be nice, but this has to be async because of the async `Response::json()`